exercises/triangle/example.cpp

```diff
@@ -1,36 +1,17 @@
 #include "triangle.h"
-#include <stdexcept>
 
-namespace triangle
+triangle::triangle(double a , double b , double c)
 {
-
-namespace
-{
-
-bool triangle_equality(double a, double b, double c)
-{
-    return a < (b + c)
-        && b < (a + c)
-        && c < (a + b);
-}
-
-}
-
-flavor kind(double a, double b, double c)
-{
-    if (a == 0 || b == 0 || c == 0) {
-        throw std::domain_error("Zero triangle");
-    }
-    if (a < 0 || b < 0 || c < 0 || !triangle_equality(a, b, c)) {
-        throw std::domain_error("Invalid triangle");
-    }
-    if (a == b && b == c) {
-        return equilateral;
-    }
-    if (a == b || b == c || a == c) {
-        return isosceles;
-    }
-    return scalene;
-}
-
+	if ( a <= 0 || b <= 0 || c <= 0 ||
```

Comment:
as the canonical data mentioned , I am returning a triangle with all properties false on error or invalid triangles instead of throwing exceptions ( as it takes less lines of code )